sql/core/src/main/scala/org/apache/spark/sql/execution/command/ddl.scala

```diff
@@ -891,8 +891,9 @@ object DDLUtils {
    * Throws exception if outputPath tries to overwrite inputpath.
    */
   def verifyNotReadPath(query: LogicalPlan, outputPath: Path) : Unit = {
-    val inputPaths = query.collect {
-      case LogicalRelation(r: HadoopFsRelation, _, _, _) => r.location.rootPaths
+    val inputPaths = EliminateBarriers(query).collect {
```

Comment:
AnalysisBarrier is a leaf node. That is one of the reasons why it could easily break the other code.